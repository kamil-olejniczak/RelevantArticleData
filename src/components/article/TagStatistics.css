.TagStatistics {
    margin: 0 auto;
    padding: 0 10px;
}

.TagStatistics tr {
    text-align: center;
}

.TagStatistics table {
    margin: 0 auto;
}

.TagStatistics td, .TagStatistics th {
    padding: 2px 10px;
}