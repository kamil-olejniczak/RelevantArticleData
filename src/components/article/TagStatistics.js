import React from 'react';

const TagStatistics = () => (
    <div>TagStatistics is working!</div>
);

export default TagStatistics;