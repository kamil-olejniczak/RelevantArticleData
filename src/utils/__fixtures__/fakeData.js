export const firstRecentlyVisitedSite = {
    id: 1,
    title: 'SomeTitle',
    url: 'SomeUrl',
    date: new Date(1 << 30),
};
export const secondRecentlyVisitedSite = {
    id: 2,
    title: 'SomeTitle2',
    url: 'SomeUrl2',
    date: new Date(1 << 30),
};
export const fakeRecentlyVisited = {
    recentlyVisited: [firstRecentlyVisitedSite, secondRecentlyVisitedSite]
};
export const fakeArticleData = {
    data: {
        title: 'Some Title',
        date_published: '1970-01-01',
        domain: 'www.dabliu.com',
        dek: 'Short dek text',
        excerpt: 'Short excerpt text',
        word_count: 111,
        lead_image_url: 'http://www.dabliu.pl/image.html',
    }
};
export const fakeTagStatistics = {
    div: 8,
    p: 2,
    a: 3,
    span: 10,
    i: 9,
    html: 1,
    br: 2,
    table: 7,
    hr: 6,
    input: 4,
    textarea: 5
};
export const fakeSortedTagStatistics = [
    'span', '10',
    'i', '9',
    'div', '8',
    'table', '7',
    'hr', '6',
    'textarea', '5',
    'input', '4',
    'a', '3',
    'br', '2',
    'p', '2',
];