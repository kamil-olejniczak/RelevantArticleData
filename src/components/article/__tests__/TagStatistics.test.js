import {shallow} from 'enzyme';
import TagStatistics from "../TagStatistics";
import React from 'react';
import renderer from 'react-test-renderer';
import {fakeSortedTagStatistics, fakeTagStatistics} from "../../../utils/__fixtures__/fakeData";

describe('TagStatistics tests', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<TagStatistics tagStats={fakeTagStatistics}/>);
    });

    it('renders correctly', () => {
        const component = renderer.create(<TagStatistics tagStats={fakeTagStatistics}/>).toJSON();

        expect(component).toMatchSnapshot();
    });

    it('should render 11 table rows, including table header', () => {
        expect(wrapper.find('tr')).toHaveLength(11);
    });

    it('should order tags by frequency of their occurrences', () => {
        const rows = wrapper.find('tr');
        const rowColumns = rows.find('td').map(column => column.text());

        rowColumns.forEach((column, index) => {
            expect(column).toEqual(fakeSortedTagStatistics[index]);
        });
    });
});