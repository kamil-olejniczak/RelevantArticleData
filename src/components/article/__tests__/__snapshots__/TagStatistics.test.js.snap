// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagStatistics tests renders correctly 1`] = `
<div
  className="TagStatistics"
>
  <span
    className="header"
  >
    Tag stats of all time:
  </span>
  <table>
    <thead>
      <tr>
        <th>
          Tag Name
        </th>
        <th>
          Occurrences
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          span
        </td>
        <td>
          10
        </td>
      </tr>
      <tr>
        <td>
          i
        </td>
        <td>
          9
        </td>
      </tr>
      <tr>
        <td>
          div
        </td>
        <td>
          8
        </td>
      </tr>
      <tr>
        <td>
          table
        </td>
        <td>
          7
        </td>
      </tr>
      <tr>
        <td>
          hr
        </td>
        <td>
          6
        </td>
      </tr>
      <tr>
        <td>
          textarea
        </td>
        <td>
          5
        </td>
      </tr>
      <tr>
        <td>
          input
        </td>
        <td>
          4
        </td>
      </tr>
      <tr>
        <td>
          a
        </td>
        <td>
          3
        </td>
      </tr>
      <tr>
        <td>
          br
        </td>
        <td>
          2
        </td>
      </tr>
      <tr>
        <td>
          p
        </td>
        <td>
          2
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
